Update IfcRelInterferesElement DocEntity & Documentation (#40)
bSI-InfraRoom/IFC-Specification: IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcRelInterferesElements/Documentation.md

> The _IfcRelInterferesElements_ objectified relationship indicates that two elements interfere. Interference is a spatial and/or shape interaction between the two elements. It is a 1 to 1 relationship. The concept of two elements interfering physically or logically is described independently from the elements. The interference may be related to the shape representation of the entities by providing an interference geometry.  

Reviewer: What is a spatial interaction and how is it different from a shape interaction?

Author: This change have being lost/reverted in actual version of the develop branch.

> * When the interference geometry is available it can be passed by the optional attribute _InterferenceGeometry_ pointing to _IfcConnectionGeometry_. The connection geometry is provided as a point, curve, surface, or volume within the local placement coordinate systems of the connecting elements. The _IfcConnectionVolumeGeometry_ is the default type to be used for interference in 3D space, as indicated in e.g. clash detections.  

Reviewer: Volume is not a term in geometry items. Volume is a measurement type. Solid is the proper terminology.

Reviewer: Not sure why I cannot add a suggestion here but this "coordinate systems of the connecting elements" is ambiguous. Which element? Only one local placement can serve as the basis. I guess Implied order should influence this also?

Reviewer 2: ```suggestion
* When the interference geometry is available it can be passed by the optional attribute _InterferenceGeometry_ pointing to _IfcConnectionGeometry_. The connection geometry is provided as a point, curve, surface, or solid within the local placement coordinate systems of the connecting elements. The _IfcConnectionVolumeGeometry_ is the default type to be used for interference in 3D space, as indicated in e.g. clash detections.  
```

Reviewer 2: @SergejMuhic amusingly your are wrong and right at the same time! :D

1.  you are correct in the sense the term is solid, but amusingly the subtype is called _IfcConnectionVolumeGeometry_ so theres that !
2. it sounds ambiguous but is correct IfcConnectionGeometry subtypes are made up of 2 geometries one for the overlapping zone on _RelatingElement_ (in its placement) and the same for _RelatedElement_ (in its placement). so...! :p

Author: Solid was kept is the actual version

Reviewer: > 
> 
> @SergejMuhic amusingly your are wrong and right at the same time! :D
> 
>     1. you are correct in the sense the term is solid, but amusingly the subtype is called _IfcConnectionVolumeGeometry_ so theres that !
> 
>     2. it sounds ambiguous but is correct IfcConnectionGeometry subtypes are made up of 2 geometries one for the overlapping zone on _RelatingElement_ (in its placement) and the same for _RelatedElement_ (in its placement). so...! :p

You have to read my comments carefully. And I would appreciate that you refrain from labeling them wrong. Thank you.

> Volume is not a term in geometry items.

That is a fact. IfcConnectionGeometry is by no means a geometric item. It has been introduced specifically to not be a geometric item (or a representation item) but to "describe the geometric and topological constraints that facilitate the physical connection of two objects". Semantics.

  
* _ImpliedOrder_=TRUE The _RelatingElement_ constitutes the primary element of the interference relationship that is oriented from _RelatingElement_ (source) to _RelatedElement_ (target). If the interference is to be resolved by subtracting the overlapping part, it should be subtracted from the _RelatingElement_. The net result would be the _RelatingElement_ subtracted by the _InterferenceGeometry_. This would be the case in interference relationships where the _RelatedElement_ creates a void in the _RelatingElement_ dynamically.  
* _ImpliedOrder_=FALSE The _RelatingElement_ and _RelatedElement_ have no priority among each other. If the interference is to be resolved then no information about whether the _InterferenceGeometry_ should be subtracted from the _RelatingElement_ or the _RelatedElement_ can be traced. This would be the case for clash detection results.  
* _ImpliedOrder_=UNKNOWN No information about the priorities is provided.  
 

> The _IfcConnectionGeometry_ property is used to define the interference shape, it can be part of the shape of one of the elements of the relationship (in the case of clash, overlapping and crossing interference types) or define the interface between the two shapes (in the case of along, over and under interference types).

Reviewer: ```suggestion
The _InterferenceGeometry_ attribute is used to define the interference shape, it can be part of the shape of one of the elements of the relationship (in the case of clash, overlapping and crossing interference types) or define the interface between the two shapes (in the case of along, over and under interference types).
```

Author: This section was lost/reverted in actual devellop version ...should it be back ?

The _InterferenceType_ property optionally specifies the type of interference between the two elements, two set of default types are provided:
* **Oriented interferences types** imply usage of _ImpliedOrder_ set to TRUE and specific choice of _RelatingElement_ and _RelatedElement_ to be meaningful:
  * Crosses: the _RelatingElement_ is crossing the _RelatedElement_ (e.g. Railway crossing a road)
  * PassesThrough: the _RelatingElement_ is passing through the _RelatedElement_ (e.g. a Road passing inside a tunnel)
  * PassesOver: the _RelatingElement_ is passing over the _RelatedElement_ (e.g a bridge passing over a water canal)

>   * PassesUnder: the _RelatingElement_ is passing under the _RelatedElement_ (e.g a Pipe passing under a road)

Reviewer: Do we allow a mix of spatial and physical concepts? I do not like this at all. @TLiebich what do you think?

> * **Non oriented interferences types** do not imply specific values of _ImpliedOrder_ (but can still be set to precise shape interference calculation)

Reviewer: ```suggestion
* **Non oriented interferences types** do not imply specific values of _ImpliedOrder_ (but can still be set to for a precise shape interference calculation)
```

>   * Clash: The _RelatingElement_ and _RelatedElement_ have a spatial or shape-based clash

Reviewer: Spatial clash has to be defined somewhere. I am having problems understanding the difference that is implied.

The _IfcRelInterferesElements_ objectified relationship indicates that two elements interfere. Interference is a spatial and/or shape interaction between the two elements. It is a 1 to 1 relationship. The concept of two elements interfering physically or logically is described independently from the elements. The interference may be related to the shape representation of the entities by providing an interference geometry.  
  
* When the interference geometry is available it can be passed by the optional attribute _InterferenceGeometry_ pointing to _IfcConnectionGeometry_. The connection geometry is provided as a point, curve, surface, or volume within the local placement coordinate systems of the connecting elements. The _IfcConnectionVolumeGeometry_ is the default type to be used for interference in 3D space, as indicated in e.g. clash detections.  
* If the interference geometry is omitted then the interference is provided as a logical relationship. Under this circumstance, the connection point, curve, surface, or solid has to be recalculated by the receiving application.  
  

> The _RelatingElement_ and _RelatedElement_ define the two elements in the relationship, that may have different roles. The relation orientation may be forcefully required by some _InterferenceType_ values, this is done  by setting the attribute _ImpliedOrder_ accordingly:

Reviewer 2: Suggestion: "The relation orientation may be required by certain _InferenceType_ values"

Author: Done

  
* _ImpliedOrder_=TRUE\S\  The _RelatingElement_ constitutes the primary element of the interference relationship that is oriented from _RelatingElement_ (source) to _RelatedElement_ (target). If the interference is to be resolved by subtracting the overlapping part, it should be subtracted from the _RelatingElement_. The net result would be the _RelatingElement_ subtracted by the _InterferenceGeometry_. This would be the case in interference relationships where the _RelatedElement_ creates a void in the _RelatingElement_ dynamically.  
* _ImpliedOrder_=FALSE\S\  The _RelatingElement_ and _RelatedElement_ have no priority among each other. If the interference is to be resolved then no information about whether the _InterferenceGeometry_ should be subtracted from the _RelatingElement_ or the _RelatedElement_ can be traced. This would be the case for clash detection results.  
* _ImpliedOrder_=UNKNOWN No information about the priorities is provided.  
 

> The _IfcConnectionGeometry_ property is used to define the interference shape, it can be part of the shape of one of the elements of the relationship (case of clash, overlapping and crossing relations) or be at the frontier of these shape (case on along, over and under relations).

Reviewer 2: Suggestion: "(in the case of clash, overlapping and crossing interference types) or define the interface between the two shapes (in the case of along, over and under interference types)."

Author: Better English taken :smile:

> The _InterferenceType_ property optionally precise the type of interferences between the two elements, two set of default types are provided:

Reviewer 2: Suggestion: "optionally specifies the type of interference between the two elements"

Author: Done

**Oriented interferences types** imply usage of _ImpliedOrder_ set to TRUE and specific choice of _RelatingElement_ and _RelatedElement_ to be meaningful:
* Crosses: the _RelatingElement_ is crossing the _RelatedElement_ (e.g. Railway crossing a road)
* PassesThrough: the _RelatingElement_ is passing through the _RelatedElement_ (e.g. a Road passing inside a tunnel)

> * PassesOver: the _RelatingElement_ is passing over the _RelatedElement_ (e.g a Pipe passing over a road)

Reviewer 2: wouldnt "bridge" be the more obvious example?

Author: Swap passingUnder and passingOver examples:

*   bridge passes over a water canal
*   pipe passes under a road
